manifests/init.pp

```diff
@@ -37,6 +37,7 @@
   Hash[String, Hash] $hosts = {},
   Variant[Array[String], Optional[String]] $includes = undef,
   String $config_comment = 'dhcpd.conf',
+  Optional[String] $service_ensure = undef,
```

Comment:
To remain compatible it should stay the same:
```suggestion
  String $service_ensure = 'running',
```

Reply:
If you would like that puppet wont change the state of the service, what value will you set?
Using undef will not change the service status, or, we can set another "if" about whether puppet should manage the service or not.
WDYT?

Reply:
Good point. How about:
```puppet
Enum['running', 'unmanaged', 'stopped'] $service_ensure = 'running',

$_service_ensure_real = $service_ensure ? {
  'unmanaged' => undef,
  default     => $service_ensure,
}
```

Reply:
Looks great :)

Reply:
Could you come up with the changes and idea also some tests?